### PR TITLE
Beta 2

### PR DESCRIPTION
**Pull request by anonymoussprocket**
The end of the beginning for ConseilJS2

### CODE REVIEW

File: src/chain/tezos/TezosMessageUtil.ts [hunk 1]
```diff
@@ -171,27 +201,91 @@ export namespace TezosMessageUtils {
...
+     * 
+     * @param {SignedOperationGroup} signedOpGroup Signed operation group
+     * @returns {string} Base58Check hash of signed operation
+     */
+    export function computeOperationHash(signedOpGroup: SignedOperationGroup): string {
+      const hash = CryptoUtils.simpleHash(signedOpGroup.bytes, 32);
+      return readBufferWithHint(hash, "op");
+    }
+
+    export function computeKeyHash(key: Buffer, prefix: string = 'tz1'): string {
```
Comment (vishakh) [inline src/chain/tezos/TezosMessageUtil.ts]: Let's document all exported functions.

File: src/chain/tezos/TezosMessageUtil.ts [hunk 2]
```diff
@@ -171,27 +201,91 @@ export namespace TezosMessageUtils {
...
+    if (hint === 'edsk') {
+      return base58check.encode(Buffer.from('2bf64e07' + key.toString('hex'), 'hex'));
+    } else if (hint === 'edpk') {
+      return readPublicKey(`00${key.toString('hex')}`);
+    } else {
+      throw new Error(`Unrecognized key hint, '${hint}'`);
+    }
+  }
+
+  export function writeKeyWithHint(b: string, hint: string): Buffer {
```
Comment (vishakh) [inline src/chain/tezos/TezosMessageUtil.ts]: Some of these new functions could have more specific names. It's not clear from the name what they do. Perhaps there should be a reference they are using Base58Check?

File: src/reporting/ConseilDataClient.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
-import fetch from 'node-fetch';
-import {ConseilServerInfo, ConseilQuery, ConseilRequestError} from "../v2/QueryTypes"
+import {ConseilServerInfo, ConseilQuery} from "../types/conseil/QueryTypes";
```
[the remark this answers is not in the document]
Comment (anonymoussprocket) [reply src/reporting/ConseilDataClient.ts]: I don’t think it hurts to be explicit. Then if something changes, there will be a build error

File: src/utils/CryptoUtils.ts [hunk 4]
```diff
@@ -1,170 +1,99 @@
...
  */
+export namespace CryptoUtils {
+    /**
+     * Generates a salt for key derivation.
+     * 
+     * @returns {Buffer} Salt
+     */
+    export function generateSaltForPwHash() {
+        let rand = Buffer.alloc(sodium.crypto_pwhash_SALTBYTES);
+        sodium.randombytes_buf(rand);
```
[the remark this answers is not in the document]
Comment (anonymoussprocket) [reply src/utils/CryptoUtils.ts]: Yea, it mirrors the cpp implementation more closely. Operation results are out params

[the remark this answers is not in the document]
Comment (anonymoussprocket) [reply src/utils/CryptoUtils.ts]: Yea, it mirrors the cpp implementation more closely. Operation results are out params

File: src/types/conseil/ErrorTypes.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import {ConseilQuery} from "./QueryTypes";
+
+export class ServiceRequestError extends Error {
```
[the remark this answers is not in the document]
Comment (anonymoussprocket) [reply src/types/conseil/ErrorTypes.ts]: Yes!

File: src/chain/tezos/TezosMessageUtil.ts (hunk 1 above)
Comment (anonymoussprocket) [reply src/chain/tezos/TezosMessageUtil.ts]: Documentation is a journey.

**Pull request merged**